### ISSUE TITLE
Breaking out fallback request interception

### ISSUE THREAD

**Issue by slightlyoff** [mkruisselbrink/navigator-connect#28]
In conversation with @jungkees, it sounds like `e.handleFallthroughRequest(...)` should probably be broken out into a separate proposal for changing the SW API.

/cc @jakearchibald @jungkees

**Comment by mkruisselbrink** [comment]
Yes, completely agree with this. Both n.c and request interception are ways to enable cross-origin service worker communication, but they are different enough for it to not really make sense to combine them in one proposal.
But what about network (http header based) installation of SWs? That probably also makes more sense as a change to the SW API rather than being part of a n.c spec?

**Comment by jakeleichtling** [comment]
I agree with breaking both `e.handleFallthroughRequest(...)` and HTTP header-based installation of SWs out of n.c, since these seem to fall into the more general SW API.

Relatedly, it may make sense to introduce a n.c flag that allows the caller to invoke the installation of the target SW if it is not already installed.

**Comment by benfrancis** [comment]
+1

**Comment by jungkees** [comment]
The fallback request interception and the HTTP header-based installation features have been proposed as extensions to the SW (v2): https://github.com/slightlyoff/ServiceWorker/issues/684, https://github.com/slightlyoff/ServiceWorker/issues/685, respectively.

**Comment by jungkees** [comment]
Just reopened it until we sort out the spec text which is now pointed to by the issues filed in SW.
